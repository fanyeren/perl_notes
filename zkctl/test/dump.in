#!/bin/bash
#
#
#
source $(dirname $0)/common

#
function prepare_env()
{
    ./zkctl -s group add --group-id=group_x --data={}
    ./zkctl -s idc add --idc-id=jx --data={}
    ./zkctl -s idc add --idc-id=tc --data={}
    ./zkctl -s slavenode add --slavenode-id=n1 --data='{"group":"group_x","idc":"jx","cpu_capacity":"100","failure_domain":"f1","online_state":"online"}'
    ./zkctl -s slavenode add --slavenode-id=n2 --data='{"group":"group_x","idc":"jx","cpu_capacity":"100","failure_domain":"f1","online_state":"online"}'
    ./zkctl -s slavenode add --slavenode-id=n3 --data='{"group":"group_x","idc":"jx","cpu_capacity":"100","failure_domain":"f1","online_state":"online"}'
    ./zkctl -s slavenode add --slavenode-id=n4 --data='{"group":"group_x","idc":"jx","cpu_capacity":"100","failure_domain":"f1","online_state":"online"}'
    ./zkctl -s slavenode add --slavenode-id=n5 --data='{"group":"group_x","idc":"tc","cpu_capacity":"100","failure_domain":"f1","online_state":"online"}'
    ./zkctl -s slavenode add --slavenode-id=n6 --data='{"group":"group_x","idc":"tc","cpu_capacity":"100","failure_domain":"f1","online_state":"online"}'
    ./zkctl -s slavenode add --slavenode-id=n7 --data='{"group":"group_x","idc":"tc","cpu_capacity":"100","failure_domain":"f1","online_state":"online"}'
    ./zkctl -s slavenode add --slavenode-id=n8 --data='{"group":"group_x","idc":"tc","cpu_capacity":"100","failure_domain":"f1","online_state":"online"}'

    ./zkctl -s index-type add --index-type=wdnb --data='{"category":"auto","trans_db_priority":"2","trans_db_timeout":"1800"}'
    ./zkctl -s index-partition add --index-partition-id=wdnb_0 --data='{"group":"group_x", "replication_factor":"1", "index_type":"wdnb", "index_layer":"1"}'
    ./zkctl -s index-partition add --index-partition-id=wdnb_1 --data='{"group":"group_x", "replication_factor":"1", "index_type":"wdnb", "index_layer":"2"}'
    ./zkctl -s index-partition add --index-partition-id=wdnb_2 --data='{"group":"group_x", "replication_factor":"1", "index_type":"wdnb", "index_layer":"3"}'
    ./zkctl -s service add --service-id=bs_wdnb_0 --data={}
    ./zkctl -s service add --service-id=bs_wdnb_1 --data={}
    ./zkctl -s service add --service-id=bs_wdnb_2 --data={}
    SID=$(./zkctl -s service-instance add --service-id=bs_wdnb_0 --slavenode-id=n1 --data='{"state":"NEW"}')
    ./zkctl -s service-instance set-current-state-all --service-instance-id=$SID --data='{"service_port":"12345"}'
    SID=$(./zkctl -s service-instance add --service-id=bs_wdnb_1 --slavenode-id=n2 --data='{"state":"NEW"}')
    ./zkctl -s service-instance set-current-state-all --service-instance-id=$SID --data='{"service_port":"12345"}'
    SID=$(./zkctl -s service-instance add --service-id=bs_wdnb_2 --slavenode-id=n3 --data='{"state":"NEW"}')
    ./zkctl -s service-instance set-current-state-all --service-instance-id=$SID --data='{"service_port":"12345"}'
    SID=$(./zkctl -s service-instance add --service-id=bs_wdnb_0 --slavenode-id=n4 --data='{"state":"NEW"}')
    ./zkctl -s service-instance set-current-state-all --service-instance-id=$SID --data='{"service_port":"12345"}'
    SID=$(./zkctl -s service-instance add --service-id=bs_wdnb_1 --slavenode-id=n5 --data='{"state":"NEW"}')
    ./zkctl -s service-instance set-current-state-all --service-instance-id=$SID --data='{"service_port":"12345"}'
    SID=$(./zkctl -s service-instance add --service-id=bs_wdnb_2 --slavenode-id=n6 --data='{"state":"NEW"}')
    ./zkctl -s service-instance set-current-state-all --service-instance-id=$SID --data='{"service_port":"12345"}'
    SID=$(./zkctl -s service-instance add --service-id=bs_wdnb_2 --slavenode-id=n7 --data='{"state":"NEW"}')
}
function clear_env()
{
    node_clear service-instance
    node_clear service
    node_clear index-partition
    node_clear index-type
    node_clear slavenode
    node_clear idc
    node_clear group
}
clear_env
prepare_env
./zkctl -s placement-plan dump 2>&1
clear_env
