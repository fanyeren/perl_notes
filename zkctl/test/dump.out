bs_wdnb_2-0000000003 get current_state "service_port" failed
{
    "index_partition":{
        "wdnb_0":{
            "group":"group_x",
            "index_layer":"1",
            "index_type":"wdnb",
            "replication_factor":"1"
        },
        "wdnb_1":{
            "group":"group_x",
            "index_layer":"2",
            "index_type":"wdnb",
            "replication_factor":"1"
        },
        "wdnb_2":{
            "group":"group_x",
            "index_layer":"3",
            "index_type":"wdnb",
            "replication_factor":"1"
        }
    },
    "placement":{
        "n1":[
            {
                "partition":"wdnb_0",
                "service_port":"12345",
                "state":"NEW"
            }
        ],
        "n2":[
            {
                "partition":"wdnb_1",
                "service_port":"12345",
                "state":"NEW"
            }
        ],
        "n3":[
            {
                "partition":"wdnb_2",
                "service_port":"12345",
                "state":"NEW"
            }
        ],
        "n4":[
            {
                "partition":"wdnb_0",
                "service_port":"12345",
                "state":"NEW"
            }
        ],
        "n5":[
            {
                "partition":"wdnb_1",
                "service_port":"12345",
                "state":"NEW"
            }
        ],
        "n6":[
            {
                "partition":"wdnb_2",
                "service_port":"12345",
                "state":"NEW"
            }
        ],
        "n7":[
            {
                "partition":"wdnb_2",
                "service_port":"-1",
                "state":"NEW"
            }
        ],
        "n8":[
        ]
    },
    "slavenode":{
        "n1":{
            "cpu_capacity":"100",
            "failure_domain":"f1",
            "group":"group_x",
            "idc":"jx",
            "online_state":"online"
        },
        "n2":{
            "cpu_capacity":"100",
            "failure_domain":"f1",
            "group":"group_x",
            "idc":"jx",
            "online_state":"online"
        },
        "n3":{
            "cpu_capacity":"100",
            "failure_domain":"f1",
            "group":"group_x",
            "idc":"jx",
            "online_state":"online"
        },
        "n4":{
            "cpu_capacity":"100",
            "failure_domain":"f1",
            "group":"group_x",
            "idc":"jx",
            "online_state":"online"
        },
        "n5":{
            "cpu_capacity":"100",
            "failure_domain":"f1",
            "group":"group_x",
            "idc":"tc",
            "online_state":"online"
        },
        "n6":{
            "cpu_capacity":"100",
            "failure_domain":"f1",
            "group":"group_x",
            "idc":"tc",
            "online_state":"online"
        },
        "n7":{
            "cpu_capacity":"100",
            "failure_domain":"f1",
            "group":"group_x",
            "idc":"tc",
            "online_state":"online"
        },
        "n8":{
            "cpu_capacity":"100",
            "failure_domain":"f1",
            "group":"group_x",
            "idc":"tc",
            "online_state":"online"
        }
    }
}
